bundles/org.openhab.binding.worxlandroid/src/main/java/org/openhab/binding/worxlandroid/internal/handler/WorxLandroidBridgeHandler.java

```diff
@@ -0,0 +1,210 @@
+import org.openhab.core.auth.client.oauth2.OAuthResponseException;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WorxLandroidBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Nils - Initial contribution
+ * @author Gaël L'hopital - Refactored with oAuthFactory, removed AWSClient
+ */
+@NonNullByDefault
+public class WorxLandroidBridgeHandler extends BaseBridgeHandler
+        implements AccessTokenRefreshListener, ThingHandlerHelper {
+    private static final String URL_OAUTH_TOKEN = "https://id.worx.com/oauth/token";
+    private static final String CLIENT_ID = "013132A8-DB34-4101-B993-3C8348EA0EBC";
+
+    private final Logger logger = LoggerFactory.getLogger(WorxLandroidBridgeHandler.class);
+    private final WorxApiHandler apiHandler;
+    private final OAuthFactory oAuthFactory;
+
+    public final OAuthClientService oAuthClientService;
+
+    private String accessToken = "";
+    private int retryCount = 3;
+    private int retryDelayS = 1;
+    private Optional<ScheduledFuture<?>> tokenRefreshJob = Optional.empty();
+    private Optional<ScheduledFuture<?>> connectionJob = Optional.empty();
+
+    public WorxLandroidBridgeHandler(Bridge bridge, WorxApiHandler apiHandler, OAuthFactory oAuthFactory) {
+        super(bridge);
+        this.apiHandler = apiHandler;
+        this.oAuthFactory = oAuthFactory;
+        this.oAuthClientService = oAuthFactory.createOAuthClientService(getThing().getUID().getAsString(),
+                URL_OAUTH_TOKEN, null, CLIENT_ID, null, "*", true);
+        oAuthClientService.addAccessTokenRefreshListener(this);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Landroid API bridge handler.");
+        WebApiConfiguration config = getConfigAs(WebApiConfiguration.class);
+
+        if (config.username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-username");
+            return;
+        }
+
+        if (config.password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-password");
+            return;
+        }
+
+        scheduler.execute(() -> initiateConnection(config.username, config.password));
+    }
+
+    private void initiateConnection(String username, String password) {
+        stopConnectionJob();
+        try {
+            accessToken = oAuthClientService.getAccessTokenByResourceOwnerPasswordCredentials(username, password, "*")
+                    .getAccessToken();
+
+            if (firstLaunch()) {
```

Comment:
With this implementation, the properties are never updated after the first launch.
You could double check, but if i remember correctly the properties will be empty each start for file based things and managed things will remember these properties.

Reply:
You're right. Here are the properties:
![image](https://github.com/user-attachments/assets/c3316ea9-ae52-4f76-b90c-4f5dfae9cd0e)

Most are not subject to change, but e.g. firmware can change. I remember having had the same discussion on another binding, and the added value of checking first run is poor. I remove it.